packages/headless/src/features/folding/folding-slice.test.ts

```diff
@@ -1,163 +1,144 @@
+import {AnyAction} from 'redux';
 import {buildMockResultWithFolding} from '../../test/mock-result-with-folding';
 import {buildMockSearch} from '../../test/mock-search';
 import {buildMockSearchResponse} from '../../test/mock-search-response';
-import {executeSearch, fetchMoreResults} from '../search/search-actions';
-import {foldingReducer} from './folding-slice';
-import {
-  FoldedResult,
-  FoldingState,
-  getFoldingInitialState,
-} from './folding-state';
-
-function getEncyclopediaCollection() {
-  const collection = 'encyclopedia';
-
-  const animalsResult = buildMockResultWithFolding({
-    uniqueId: '_animals',
-    raw: {urihash: '', collection, id: ['animals']},
-  });
-
-  const ballPythonsResult = buildMockResultWithFolding({
-    uniqueId: '_ballPythons',
-    raw: {urihash: '', collection, parent: 'reptiles'},
-  });
+import {executeSearch} from '../search/search-actions';
+import {foldingReducer, ResultWithFolding} from './folding-slice';
+import {FoldedResult, FoldingState} from './folding-state';
 
-  const geckosResult = buildMockResultWithFolding({
-    uniqueId: '_geckos',
-    raw: {urihash: '', collection, parent: 'reptiles'},
-  });
+interface Hierarchy {
+  name: string;
+  children?: Hierarchy[];
+}
 
-  const mammalsResult = buildMockResultWithFolding({
-    uniqueId: '_mammals',
+function buildMockResultsFromHierarchy(
```

Comment:
I created a bunch of utils for folding unit tests. I originally wanted to avoid them, but folding unit tests without all three of these would be extremely hard to read and would take way too much code.

This first util creates a flat array of results from a recursive `Hierarchy` interface that is much easier to read/write.